src/tesla_client.cpp

```diff
@@ -0,0 +1,745 @@
+	bool response_charge_port_door_open = jresponse["charge_port_door_open"]; // true
+	const char* response_charge_port_latch = jresponse["charge_port_latch"]; // "Engaged"
+	int response_charge_rate = jresponse["charge_rate"]; // 0
+	bool response_charge_to_max_range = jresponse["charge_to_max_range"]; // false
+	int response_charger_actual_current = jresponse["charger_actual_current"]; // 0
+	int response_charger_pilot_current = jresponse["charger_pilot_current"]; // 48
+	int response_charger_power = jresponse["charger_power"]; // 0
+      
+	const char* response_charging_state = jresponse["charging_state"]; // "Complete"
+	const char* response_conn_charge_cable = jresponse["conn_charge_cable"]; // "IEC"
+	float response_est_battery_range = jresponse["est_battery_range"]; // 187.27
+	const char* response_fast_charger_brand = jresponse["fast_charger_brand"]; // ""
+	bool response_fast_charger_present = jresponse["fast_charger_present"]; // false
+	const char* response_fast_charger_type = jresponse["fast_charger_type"]; // ""
+	float response_ideal_battery_range = jresponse["ideal_battery_range"]; // 230.21
+	bool response_managed_charging_active = jresponse["managed_charging_active"]; // false
+	bool response_managed_charging_user_canceled = jresponse["managed_charging_user_canceled"]; // false
+	int response_max_range_charge_counter = jresponse["max_range_charge_counter"]; // 0
+	int response_minutes_to_full_charge = jresponse["minutes_to_full_charge"]; // 0
+	bool response_not_enough_power_to_heat = jresponse["not_enough_power_to_heat"]; // false
+	bool response_scheduled_charging_pending = jresponse["scheduled_charging_pending"]; // false
+      
+	long response_timestamp = jresponse["timestamp"]; // 1583079530271
+	bool response_trip_charging = jresponse["trip_charging"]; // false
+	int response_usable_battery_level = jresponse["usable_battery_level"]; // 83
+#endif // notyet
+      }
+      _activeRequest = TAR_NONE;
+    });
+  _client.send(req);
+}
+
+#endif // USE_WFCS
+
+void TeslaClient::getChargeInfoJson(String &sjson)
+{
+  DynamicJsonDocument doc(JSON_OBJECT_SIZE(7));
+
+  if (_chargeInfo.isValid) {
+    doc["tesla/batteryRange"] = _chargeInfo.batteryRange;
+    doc["tesla/chargeEnergyAdded"] = _chargeInfo.chargeEnergyAdded;
+    doc["tesla/chargeMilesAddedRated"] = _chargeInfo.chargeMilesAddedRated;
+    doc["tesla/batteryLevel"] = _chargeInfo.batteryLevel;
+    doc["tesla/chargeLimitSOC"] = _chargeInfo.chargeLimitSOC;
+    doc["tesla/timeToFullCharge"] = _chargeInfo.timeToFullCharge;
+    doc["tesla/chargerVoltage"] = _chargeInfo.chargerVoltage;
+    serializeJson(doc,sjson);
+  }
+  else {
+    sjson = "";
+  }
+}
+
+#ifdef TESLAERRORS
+503 Service Unavailable
+_HEADER[Server]: BigIP
+_HEADER[Connection]: Keep-Alive
+_HEADER[Content-Length]: 95
+
+<html><head></head><body><p>Our servers are waiting for a supercharger spot..</p></body></html>
+token error
+
+#endif
```

Comment:
No, I left it in by accident. it was the start of a log of various errors the server can return. the code doesn't handle all the possible error scenarios, because I don't know what they are